### ISSUE TITLE
Only static rain available?

### ISSUE THREAD

**Issue by jwgrlrrajn** [07th-mod/umineko-question#59]
Granted, I haven't really tested this extensively, but I noticed that in the original UmiTweak patch for the pre-MG release of the game, the rain was actually animated. However, so far, I've only seen static rain images used while jumping around the game. Is the static rain something that was actually in the game the whole time, with the animated rain being an UmiTweak thing? If so, do we know how we could possibly replicate the animated rain?

**Comment by AntonStrickland** [comment]
Animated rain was in the original game, in both NScripter and ONScripter releases. When MG released it on Steam, they switched the engine to PONScripter, in which dynamic rain does not work. I have no idea how you would need to modify PONScripter to get it working again, though. Probably something to do with the "layermessage" function from looking at the original game's script.

**Comment by jwgrlrrajn** [comment]
Hmm. I wonder if PONscripter supports APNG's. If I can get one of those to animate properly in the engine, it might be an option, albeit a hacky one. APNG's support transparency, so the only concern is if the image will behave correctly in the engine. I'll test it later and see what I can come up with.

**Comment by jwgrlrrajn** [comment]
I tested for APNG support. Sadly, no dice. I wonder why the switch to PONscripter was made to begin with if it didn't support one of the most often reoccurring aesthetics?

**Comment by ItaloKnox** [comment]
Simple: multiplatform and multilanguage. I always thought the game was going to receive the Higurashi treatment but it was just a half-assed port.

**Comment by drojf** [comment]
Did you try using the animation function/alpha function of the `lsp` function? To display an animated sprite on top of whatever is on the screen.

http://unclemion.com/onscripter/api/NScrAPIm.xml#lsp

specifically: ~~"a - for 'alphablend'; the right half of the image contains alpha-level data for the left half"~~
"The image gets 'cut' into (num-cells) pieces, going from left to right. If the image contains alpha-level data, then each cell contains its own alpha-data in its right-half."

So it seems to use a 'spritesheet' method for animation - you would have copies of the image from left to right forming a very long image (this is explained in the lsp description)

~~For alpha, you make a double image, with the left image the color, and the right image the alpha values (as greyscale I assume). I think there are some examples of this in the "sprite effects" and in the backgrounds of the game (the prison cell background uses that kind of transparency).~~

So if you want alpha and animation, I assume that you would have a series of ~~double~~ images, the combination of the above two (...if that even works).

This is all guesswork and might not be supported by the engine. I'm not sure if it's worth putting effort into it but you can try if you like. If you scan through the script there should be some examples of transparency, but not sure if there are any examples of animation.

In other words:
- check alpha (see if proper transparency is supported, not just on/off transparency)
- ~~check animation~~
- ~~check alpha + animation~~
- check alpha + animation ontop of other things on the screen.

EDIT:

Looks like "cursor.png" is an animated image with alpha, see:

```
lsph TextCC1SpNum,":bl/3,160,2;bmp\cursor0.png",0,0
lsph TextCC2SpNum,":bl/3,160,2;bmp\cursor1.png",0,0
```

in the script. It appears you can just use .PNG files and don't need to use the double image method for alpha. Since that's already proven, it's worth trying out. But actually incorporating it in the script may take a lot of effort, even if it works. The .png files are attached, but I think it is provided in the graphics patch anyway.

![cursor1](https://cloud.githubusercontent.com/assets/1249449/26730999/ff87ebac-47f5-11e7-87d4-3e99ee8da70c.png)
![cursor0](https://cloud.githubusercontent.com/assets/1249449/26731000/ff88f43e-47f5-11e7-942a-5087f5704491.png)

**Comment by jwgrlrrajn** [comment]
I know this is low priority due to work on Chiru, but has there been any progress on this since the last post?

**Comment by drojf** [comment]
no...I don't think this will be worked on until after Chiru is finished. I can give advice if you want to work on on it or start the ball rolling though. 


may also be worth joining the discord channel to see where we're at: https://discord.gg/acSbBtD

**Comment by drojf** [comment]
I felt like it so I did work on it a little bit. 

For future reference:

`_lsp 0,":bl/4,30,2;cursor0.png",0,0`

The '/4' part means, there are 4 frames in the animation. You don't need to specify the width or the height - it will assume it's 4 images laid out from left to right (a very wide image).

The next argument, '30' is how long each frame will be displayed.

Now, the issue with this method is that each time you click to display a new line of text, the rain freezes while the line displays (as it types itself out), which is pretty distracting. I don't think this method is really usable because of this :(.

So unless while patching chiru I discover some alternate method, I don't think this will work (well, technically setting the text to display instantaneously does work...). 

I'll leave this thread open just in case I figure something out.

----

Here is the image I used: [cursor0](https://user-images.githubusercontent.com/1249449/34079821-58dc4444-e389-11e7-9dc3-2456b03f74cf.png)

**Comment by drojf** [comment]
See: https://github.com/07th-mod/umineko-question/issues/126
